extensions/amp-live-list/0.1/poller.js

```diff
@@ -0,0 +1,186 @@
+  constructor(win, wait, work) {
+    /** @private @const {!Window} */
+    this.win = win;
+
+    /** @private {number} */
+    this.wait_ = wait;
+
+    /** @private {function(): !Promise} */
+    this.work_ = work;
+
+    /** @private {?number} */
+    this.lastTimeoutId_ = null;
+
+    /** @private {?number} */
+    this.lastBackoffTimeoutId_ = null;
+
+    /** @private {function(): number} */
+    this.backoffPoller_ = this.getBackoffPoller_();
+
+    /** @private {!PollerState} */
+    this.state_ = PollerState.STOPPED;
+
+    /**
+     * Mostly for testing purposes.
+     * @private {!Promise}
+     */
+    this.lastWorkPromise_ = null;
+  }
+
+  /**
+   * @return {function(): number}
+   * @private
+   */
+  getBackoffPoller_() {
+    return exponentialBackoff();
+  }
+
+  /**
+   * @return {!PollerState}
+   */
+  getState() {
+    return this.state_;
+  }
+
+  /**
+   * Initalize any work needed to start polling.
+   */
+  start() {
+    if (this.lastTimeoutId_) {
+      return;
+    }
+    if (this.lastBackoffTimeoutId_) {
+      this.win.clearTimeout(this.lastBackoffTimeoutId_);
+    }
+
+    // we reset the backoff poller
+    this.backoffPoller_ = this.getBackoffPoller_();
+    this.state_ = PollerState.RUNNING;
+    this.poll_();
+  }
+
+  /**
+   * Initialize backoff function and set correct state.
+   */
+  backoff() {
+    // if theres already backoff work queued, then cancel this invocation.
+    if (this.lastBackoffTimeoutId_) {
+      return;
+    }
+    this.stop();
+
+    this.state_ = PollerState.BACKEDOFF;
+    this.backoff_();
+  }
+
+  /**
+   * Clear timers and set correct stop state.
+   */
+  stop() {
+    this.state_ = PollerState.STOPPED;
+    if (this.lastTimeoutId_) {
+      this.win.clearTimeout(this.lastTimeoutId_);
+      this.lastTimeoutId_ = null;
+    }
+    if (this.lastBackoffTimeoutId_) {
+      this.win.clearTimeout(this.lastBackoffTimeoutId_);
+      this.lastBackoffTimeoutId_ = null;
+    }
+  }
+
+  /**
+   * @private
+   */
+  backoff_() {
+    if (this.state_ != PollerState.BACKEDOFF) {
+      return;
+    }
+    this.lastBackoffTimeoutId_ = this.backoffPoller_(() => {
+      this.lastWorkPromise_ = this.work_().then(this.backoff_.bind(this))
+          .catch(e => {
+            this.backoff_();
+            throw e;
+          });
+    });
+  }
+
+  /**
+   * @private
+   */
+  poll_() {
+    if (this.state_ != PollerState.RUNNING) {
+      return;
+    }
+    const wait = Poller.getWaitTime_(this.wait_);
+    this.lastTimeoutId_ = this.win.setTimeout(() => {
+      // Whether the work fails or passes, poll again. We need to be told
+      // externally to stop.
+      this.lastWorkPromise_ = this.work_().then(this.poll_.bind(this))
+          .catch(e => {
+            this.poll_();
+            throw e;
+          });
+    }, wait);
+  }
+
+  /**
+   * Get wait time + jitter.
+   * @return {number}
+   * @static
+   */
+  static getWaitTime_(wait) {
+    let jitter = wait * .2 * Math.random();
```

Comment:
You can just do: `jitter = wait * 0.2 * (0.5 - Math.random())` to get both directions.